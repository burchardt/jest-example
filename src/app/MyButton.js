import React from "react";

export class MyButton extends React.Component {
    constructor(props) {
        super(props);
        this.state = {open: true}
    }

    toggleState() {
        this.setState({open : !this.state.open});
    };

    getButtonText() {
       return this.state.open ? "Open" : "Close";
    }

    render() {
        const buttonStyle = {
            "fontSize": "25px",
            "backgroundColor": "lightblue"
        };

        return(
            <div>
                <button id="myButton"
                    className="btn"
                    style={buttonStyle}
                    onClick={this.toggleState.bind(this)}
                >
                {this.getButtonText()}
                </button>
            </div>
        );
    }
}