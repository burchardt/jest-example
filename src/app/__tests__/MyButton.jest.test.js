import React from 'react';
import renderer from 'react-test-renderer';

import { MyButton } from '../MyButton';

describe('<Button />', () => {
    it('should render a button with a classname', () => {
        const tree = renderer.create(
            <MyButton />
        ).toJSON();
        // Generate and match snapshot
        expect(tree).toMatchSnapshot();
    });

    it('toggleState should be called on click', () => {
        const component = renderer.create(
            <MyButton />
        );
        let tree = component.toJSON();
        // Find the button element
        const button = tree.children.filter(child => child.type === 'button')[0];
        // Check if the button were found and is not null
        expect(button).toBeDefined();
        expect(button).not.toBeNull();
        // Simulate click on button
        button.props.onClick();
        // Generate snapshot after the button was clicked
        tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});