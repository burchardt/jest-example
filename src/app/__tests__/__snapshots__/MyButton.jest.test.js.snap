// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render a button with a classname 1`] = `
<div>
  <button
    className="btn"
    id="myButton"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "lightblue",
        "fontSize": "25px",
      }
    }
  >
    Open
  </button>
</div>
`;

exports[`<Button /> toggleState should be called on click 1`] = `
<div>
  <button
    className="btn"
    id="myButton"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "lightblue",
        "fontSize": "25px",
      }
    }
  >
    Close
  </button>
</div>
`;
